src/v/cluster/types.cc

```diff
@@ -136,7 +136,8 @@ bool topic_properties::has_overrides() const {
            || read_replica.has_value() || batch_max_bytes.has_value()
            || retention_local_target_bytes.has_value()
            || retention_local_target_ms.has_value()
-           || remote_delete != storage::ntp_config::default_remote_delete;
+           || remote_delete != storage::ntp_config::default_remote_delete
+           || segment_ms.has_value();
```

Comment:
Don't we need to check `is_disabled()` here? I.e. if a user explicitly set the topic property to -1, we should honor that, rather than using the cluster-wide value, right?

Reply:
Could you add a test similar to `ShadowIndexingInfiniteRetentionTest` that exercises the -1 case?

Reply:
This smells like a bug, but it's an odd one. If the user has set `retention_local_target_*=-1`, but _hasn't_ set the `retention_*` values, it's a bit unclear what the behavior should be. Regardless, it feels incorrect to ignore the -1.

@jcsp @VladLazar thoughts?

Reply:
I find the `tristate` type confusing at the best of times.  This seems worth tracking, can you open an issue so that we don't forget about it next week?  Vlad probably has the best context.

Reply:
https://github.com/redpanda-data/redpanda/issues/8344

Reply:
I agree with Andrew that it is a bug. However, it will never manifest for Kafka topics as we always override the cleanup policy (see code [here](https://github.com/redpanda-data/redpanda/blob/0ef2100f72189652b758155cd66501354d12eba3/src/v/kafka/server/handlers/create_topics.cc#L247-L258)). Still worth fixing. I'll put up a PR.